scripts/globals/artisan.lua

```diff
@@ -0,0 +1,86 @@
+-----------------------------------
+--
+--  Artisan Moogles
+--
+-----------------------------------
+require('scripts/globals/zone')
+require('scripts/globals/status')
+-----------------------------------
+
+tpz = tpz or {}
+tpz.artisan = tpz.artisan or {}
+
+local event = {
+    [tpz.zone.BASTOK_MARKETS] = 544,
+    [tpz.zone.WINDURST_WOODS] = 833,
+    [tpz.zone.RULUDE_GARDENS] = 10162,
+    [tpz.zone.SOUTHERN_SAN_DORIA] = 960
+}
+local menuFlags = {
+	expand = 0x8,
+	abags = 0x4,
+	aexpand = 0x2,
+}
+
+tpz.artisan.moogleOnTrigger = function(player, npc)
+    local csid = event[player:getZoneID()]
+    local menuMask = 0
+    local mogVisited = player:getCharVar("[artisan]visited")
```

Comment:
Let's make this mogVisited assignment if the sackSize is > 0 or the var is one. Then when the player purchases a sack (probably every player will), we can nuke the charVar, and not have a forever var lingering in char_vars table.

Reply:
note: in other similar cases where you can't avoid what would be a forever var, you start newly created characters at 1 and then zero them after, instead of starting 0 and moving to one. (see existing examples in player.lua)

Here, you don't have to do player.lua because we can check what ibm said.

Reply:
Great idea, ibm. I've changed it so it'll only be set 1 for as long as the player has no sack, and has talked to a moogle. It will be zeroed once they have a bag and check will be based on either or.

I'll push it once I've tested a bit later.